docs/HowtoTDE.md

```diff
@@ -0,0 +1,219 @@
+
+The `tde.json` configuration file can be injected using the following Helm values file snippet:
+
+```yaml
+...
+
+admin:
+  podAnnotations:
+    vault.hashicorp.com/agent-inject: "true"
+    vault.hashicorp.com/role: nuodb
+    vault.hashicorp.com/agent-inject-file-demo: tde.json
+    vault.hashicorp.com/secret-volume-path-demo: /etc/nuodb/tde/demo
+    vault.hashicorp.com/agent-inject-secret-demo: v2.nuodb.com/TDE/demo
+    vault.hashicorp.com/agent-inject-template-demo: |
+      {{- with secret "v2.nuodb.com/TDE/demo" -}}
+      {{- .Data.data | toJSON }}
+      {{- end -}}
+ 
+database:
+  podAnnotations:
+    vault.hashicorp.com/agent-inject: "true"
+    vault.hashicorp.com/role: nuodb
+    vault.hashicorp.com/agent-inject-file-demo: tde.json
+    vault.hashicorp.com/secret-volume-path-demo: /etc/nuodb/tde/demo
+    vault.hashicorp.com/agent-inject-secret-demo: v2.nuodb.com/TDE/demo
+    vault.hashicorp.com/agent-inject-template-demo: |
+      {{- with secret "v2.nuodb.com/TDE/demo" -}}
+      {{- .Data.data | toJSON }}
+      {{- end -}}
+
+...
+```
+
+Templates for Vault KV store version 1 and version 2 are different..
+If Vault KV store version 1 is used, `.Data.data` references in inject templates should be replaced with `.Data`.
+
+> **NOTE**: For information on how different annotations can be used, see [here](https://www.vaultproject.io/docs/platform/k8s/injector/annotations).
+
+If `tde.conf` needs to be created using several secrets or special keys from the secret needs to be filtered out, [scratch](https://github.com/hashicorp/consul-template#scratch) Consul template API functions can be used. 
+For example this template will filter out any other keys in a Vault secret:
+
+```yaml
+    vault.hashicorp.com/agent-inject-template-demo: |
+      {{- with secret "v2.nuodb.com/TDE/demo" -}}
+      {{- range $k, $v := .Data.data -}}
+      {{- if or (eq $k "target") ($k | regexMatch "historical.*") }}
+      {{- scratch.MapSet "passwords" $k $v }}
+      {{- end }}
+      {{- end }}
+      {{- scratch.Get "passwords" | toJSON }}
+      {{- end -}}
+```
+
+
+## Enable Transparent Data Encryption
+
+NuoDB Admin will configure supplied storage passwords and will propagate them to engines.
+To verify that the target password is configured correctly, the following command should succeed returning no errors.
+
+```bash
+kubectl exec admin-nuodb-cluster0-0 -- nuocmd check data-encryption \
+  --db-name demo \
+  --password 'topSecret'
+```
+
+Now that the database has a storage password, encryption can be enabled using `ALTER DATABASE CHANGE ENCRYPTION TYPE`. Database administrator privileges are required in order to use this command.
+The SQL statement should be executed manually by obtaining SQL connection towards the target database.
+
+```bash
+kubectl exec -n nuodb admin-nuodb-cluster0-0 -- bash -c \
+  'echo "alter database change encryption type AES128;" | /opt/nuodb/bin/nuosql demo --user dba --password secret'
```

Comment:
this is a really nitpicky comment, but should the SQL statement be in CAPS?

Reply:
I also think the SQL will look better in CAPs. If anything, it will stand out better.